### PR TITLE
feat(shared-react): update shared plugin

### PR DESCRIPTION
**Pull request by debsmita1**
**Resolves:**
https://github.com/janus-idp/backstage-plugins/issues/539

Updated a shared component
Added more tests

### CODE REVIEW

**Comment by invincibleJai** [comment]
@debsmita1 Thanks for PR, pls link ticket to this

File: plugins/shared-react/src/components/pipeline/HorizontalStackedBars.tsx [hunk 1]
```diff
@@ -10,23 +10,32 @@ type StackedValue = {
...
+  values: StackedValue[];
+  width?: number | string;
+  onClick?: () => void;
+}) => {
+  if (onClick) {
+    const element = document.getElementById(`horizontal-stacked-bars-${id}`);
+    element?.addEventListener('click', () => {
+      onClick();
+    });
+  }
```
Comment (invincibleJai) [inline plugins/shared-react/src/components/pipeline/HorizontalStackedBars.tsx:35]: can't we add onClick directly on div?

File: plugins/shared-react/src/components/pipeline/HorizontalStackedBars.tsx [hunk 2]
```diff
@@ -10,40 +10,42 @@ type StackedValue = {
...
+  onClick,
+}: {
+  id: string;
+  barGap?: number;
+  height?: number | string;
+  inline?: boolean;
+  values: StackedValue[];
+  width?: number | string;
+  onClick?: () => void;
+}) => (
```
Comment (divyanshiGupta) [inline plugins/shared-react/src/components/pipeline/HorizontalStackedBars.tsx:29]: @debsmita1 please create a type with these props.

**Review by divyanshiGupta — CHANGES_REQUESTED** [review]

**Review by divyanshiGupta — APPROVED** [review]
LGTM!

**Pull request merged**